dpctl/tensor/_ctors.py

```diff
@@ -1110,6 +1111,16 @@ def full(
             sycl_queue=sycl_queue,
         )
         return dpt.copy(dpt.broadcast_to(X, shape), order=order)
+    # TODO: verify if `np.True_` and `np.False_` should be instances of
+    # Number in NumPy, like other NumPy scalars and like Python bools
+    # check for `np.bool_` separately as NumPy<2 has no `np.bool`
+    elif not isinstance(fill_value, Number) and not isinstance(
+        fill_value, np.bool_
+    ):
+        raise TypeError(
+            "`full` array cannot be constructed with value of type "
+            f"{type(fill_value)}"
+        )
```

Comment:
It would be nice to change this to 

```
else:
     _validate_fill_value(fill_value)
```

and result `_validate_fill_value` function in both `full` and `full_like` implementation.